legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/api/QueryStoreManager.java

```diff
@@ -200,17 +197,22 @@ public static void validateQuery(Query query)
         // TODO: we can potentially create a pattern check for version
     }
 
+    public List<Query> getQuerySearchResults(QuerySearchSpecification searchSpecification, List<Bson> filters)
+    {
+        return LazyIterate.collect(this.getQueryCollection()
+                .find(filters.isEmpty() ? EMPTY_FILTER : Filters.and(filters))
+                .sort(searchSpecification.showLatestQueriesFirst != null && searchSpecification.showLatestQueriesFirst ? descending("lastUpdatedAt") : EMPTY_FILTER)
```

Comment:
Like said, if we are already opinionated about how we sort the list of queries (i.e. returning user queries first), we can remove `showLatestQueriesFirst` flag, I don't see why we need that anymore